activerecord/lib/active_record/errors.rb

```diff
@@ -123,6 +123,28 @@ class RecordNotUnique < WrappedDatabaseException
   class InvalidForeignKey < WrappedDatabaseException
   end
 
+  # Raised when a foreign key constraint cannot be added because the column type does not match the referenced column type.
+  class MismatchedForeignKey < WrappedDatabaseException
+    def initialize(message)
+      parts = message.scan(/`(\w+)`[ $)]/).flatten
```

Comment:
I agree this is brittle.  The original implementation simply bubbled up the DBMS error, but this was added by request from Sean.  In order to display the column and table names from the error -- and provide the specific message that was requested -- this was the best way I could come up with.  I'm happy to back it out, but then need direction on the error message.